File: C#1/04-Console-IO-Homework/06QuadraticEquation/QuadraticEquation.cs
using System;
using System.Threading;
using System.Globalization;

//Write a program that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 and 
//solves it (prints its real roots).

//Input
//•On the first three lines, you will receive the coefficients a, b, and c, each on a separate line 
//in the same order

//Output
//•If two different real roots exist, print them on two separate lines 
//◦Print the smaller root on the first line

//•If only one real root exists, print it on the only output line
//•If no real root exists, print the string "no real roots"
//•The roots, should they exist, must be printed with precision exactly two digits after the floating point

//Constraints
//•The input will always consist of valid real numbers in the range [-1000, 1000] and will follow the 
//described format



class QuadraticEquation
{
        static void Main()
        {
            Thread.CurrentThread.CurrentCulture = CultureInfo.InvariantCulture;

            float a = float.Parse(Console.ReadLine());
            float b = float.Parse(Console.ReadLine());
            float c = float.Parse(Console.ReadLine());

            double x1, x2, x;

            double discriminant = b * b - 4 * a * c;

            if (discriminant > 0)
            {
                //2 real roots
                x1 = (-b + Math.Sqrt(discriminant)) / (2 * a);
                x2 = (-b - Math.Sqrt(discriminant)) / (2 * a);
                Console.WriteLine((x1 < x2) ? "{0:f2}\n{1:f2}" : "{1:f2}\n{0:f2}", x1, x2);

            }
            else if (discriminant == 0)
            {
                //1 real root
                x = -b / (2 * a);
                Console.WriteLine("{0:f2}",x);
            }
            else
            {
                //no real roots
                Console.WriteLine("no real roots");
            }            
        }
}

File: JS/HW/Conditional-Statements/quadratic-equation.js
//Write a script that reads the coefficients a, b and c of a quadratic equation ax2 + bx + c = 0 
//and solves it (prints its real roots). Calculates and prints its real roots.
//Note: Quadratic equations may have 0, 1 or 2 real roots.

//The output should be a single line containing the real roots (see sample tests)
//Print numbers with two digits of precision after the floating point
//If there are two roots then x1 < x2

function solve(args){
    var a = +args[0];
    var b = +args[1];
    var c = +args[2];

    var d = b*b - 4*a*c;

    if(d < 0){
        console.log("no real roots");
    }else if(d === 0){
        var x = -b/(2*a);
        console.log("x1=x2=" + x.toFixed(2));
    }else{
        var x1 = (-b + Math.sqrt(d))/(2 * a);
        var x2 = (-b - Math.sqrt(d))/(2 * a);

        if(x1 < x2){
            //x1=0.00; x2=3.00
            console.log("x1="+x1.toFixed(2)+"; x2=" + x2.toFixed(2));
        }else{
            console.log("x1="+x2.toFixed(2)+"; x2=" + x1.toFixed(2));
        }
    }
}

solve(['2', '5', '-3']);
solve(['-1', '3', '0']);
solve(['-0.5', '4', '-8']);
solve(['5', '2', '8']);
solve(['0.2', '9.572', '0.2']);